clientlibrary/worker/worker.go

```diff
@@ -286,13 +386,18 @@ func (w *Worker) eventLoop() {
 
 				// log metrics on got lease
 				w.mService.LeaseGained(shard.ID)
-
-				log.Infof("Start Shard Consumer for shard: %v", shard.ID)
-				sc := w.newShardConsumer(shard)
+				var shardConsumer interface{ getRecords() error }
```

Comment:
create a common interface for both consumer and put it under: https://github.com/vmware/vmware-go-kcl/tree/master/clientlibrary/interfaces